pkg/kubeapiserver/options/authentication.go

```diff
@@ -277,7 +278,16 @@ func (s *BuiltInAuthenticationOptions) ToAuthenticationConfig() authenticator.Au
 	}
 
 	if s.ClientCert != nil {
-		ret.ClientCAFile = s.ClientCert.ClientCA
+		if len(s.ClientCert.ClientCAFile) > 0 {
+			var err error
+			ret.ClientCAData, err = ioutil.ReadFile(s.ClientCert.ClientCAFile)
+			if err != nil {
+				return authenticator.AuthenticatorConfig{}, nil
+			}
+
+		} else {
+			ret.ClientCAData = s.ClientCert.ClientCAData
```

Comment:
Does it make sense if we have both a ca file and passed in ca data to give priority to the file? Better yet can we make sure that both are not set? Why don't we just want them to push the byte array to the FS?